### PR TITLE
Clarify batchworld and params comments

### PR DESCRIPTION
**Pull request by klshuster**

### CODE REVIEW

File: parlai/core/worlds.py [hunk 1]
```diff
@@ -827,6 +827,8 @@ def create_task(opt, user_agents):
     # Single threaded or hogwild task creation (the latter creates multiple threads).
     # Check datatype for train, because we need to do single-threaded for
     # valid and test in order to guarantee exactly one epoch of training.
+    # If batchsize is > 1, default to BatchWorld, regardless of numthreads, as
+    # numthreads may be used for a threadpool in BatchWorld.
```
Comment (alexholdenmiller) [inline parlai/core/worlds.py]: "used for multithreaded data loading" is mos re clear I think

**Comment by jaseweston** [comment]
but couldn't it be used for other multithreaded stuff in the long-term even
during batch mode, or no?
it might be better to write for general case (and give example)

**Comment by klshuster** [comment]
so something like "If batchsize > 1, default to BatchWorld, as numthreads can be used for multithreading in batch mode, e.g. multithreaded data loading"?

**Comment by jaseweston** [comment]
sounds good to me..alex?

**Comment by alexholdenmiller** [comment]
we may want to also just add another parameter--'hogwild' :)

**Pull request merged**